### ISSUE TITLE
Add hook allowing to prevent certain plugins archiving

### ISSUE THREAD

**Issue by mgazdzik** [piwik/piwik#10367]
Goal of this issue is to add functionality allowing to prevent certain plugins archiving on the fly (during archiver run).

Currently it is only possible to prevent given plugin archiving by disabling it entirely.
However for performance reasons it might be beneficial to prevent heavy plugin archiving under some circumstances to allow archiving run till the end (for ex. prevent actions plugin archiving, but allow remaining to process their data).
Basically such decision could be done based on plugin name and archiving parameters.

After core code review I have found no possibility to do this in nice way but to add new hook in PluginsArchiver class. 

MR for that feature is available at https://github.com/piwik/piwik/pull/10366

Please let me know if such approach is acceptable in core, or should such feature be provided in different way?

**Comment by mattab** [comment]
Done in #10366
